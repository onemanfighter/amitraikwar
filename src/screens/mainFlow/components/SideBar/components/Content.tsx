import {
  AbsoluteCenter,
  Box,
  Divider,
  IconButton,
  Img,
  Text,
} from '@chakra-ui/react';
import { TypeWriterText, useColorSelector } from '@components';
import { CardProps } from './types';
import { useTranslation } from 'react-i18next';
import SocialButtons from './SocialButtons';
import { BirthdayIcon, EmailIcon, LocationIcon, PhoneIcon } from '@assets';
import packageJson from '../../../../../../package.json';
import Profile from '@assets/images/AmitRaikwar.png';

const Icons = ({ icon }: { icon: string }) => {
  switch (icon) {
    case 'birthday':
      return <BirthdayIcon />;
    case 'email':
      return <EmailIcon />;
    case 'phone':
      return <PhoneIcon />;
    case 'location':
      return <LocationIcon />;
    default:
      return <></>;
  }
};

const Card = ({ title, heading, icon }: CardProps) => {
  const { bg, icon: iconColor } = useColorSelector();

  return (
    <Box
      padding={'4'}
      borderRadius={'lg'}
      borderBottom={'1px solid gray'}
      display={'flex'}
      flexDirection={'row'}
      justifyContent={'flex-start'}
      bg={bg.container}
    >
      <IconButton
        variant={'outline'}
        bg={iconColor.primary.bg}
        color={iconColor.primary.color}
        aria-label={title}
        icon={<Icons icon={icon} />}
      />
      <Box display={'flex'} flexDirection={'column'} paddingX={2}>
        <Text fontSize={'sm'}>{heading}</Text>
        <Text fontSize={'md'} fontWeight={'600'}>
          {title}
        </Text>
      </Box>
    </Box>
  );
};

const Content = () => {
  const { t } = useTranslation();
  const { bg, gradient } = useColorSelector();

  return (
    <Box
      flex={1}
      flexDirection={'column'}
      justifyContent={'flex-start'}
      borderRadius={'lg'}
      bgGradient={gradient.sideBarBG}
      marginX={'4'}
      marginTop={'40'}
      dropShadow={'lg'}
    >
      <Box
        boxSize={'250'}
        bg={bg.container}
        position={'relative'}
        top={'-125'}
        marginX={'auto'}
        borderRadius={'10'}
        boxShadow={'lg'}
      >
        <Img
          boxSize={'250'}
          src={Profile}
          loading="lazy"
          borderRadius={10}
          dropShadow={'lg'}
          border={'1px solid gray'}
        />
      </Box>
      <Box
        display={'flex'}
        flexDirection={'column'}
        padding={'4'}
        borderRadius={'lg'}
        bg={bg.container}
        boxShadow={'lg'}
        margin={'4'}
        position={'relative'}
        top={'-90'}
      >
        <TypeWriterText
          fixedText="I'm a "
          textSequence={[
            {
              text: 'Software engineer!',
              wait: 1000,
            },
            { text: 'Full-stack developer!', wait: 2000 },
            { text: 'Web developer!', wait: 2000 },
            { text: 'Mobile developer!', wait: 2000 },
          ]}
          styles={{
            fontSize: '1.3em',
            fontWeight: 'bold',
            fontFamily: 'monospace',
            textAlign: 'center',
          }}
        />
      </Box>
      <Box
        display={'flex'}
        flexDirection={'column'}
        padding={'4'}
        gap={'4'}
        borderRadius={'lg'}
        bg={bg.container}
        boxShadow={'xl'}
        margin={'4'}
        position={'relative'}
        top={'-90'}
      >
        <Card
          title={t('MainHomeScreen.sidebar.details.phone')}
          heading={t('MainHomeScreen.sidebar.details.phoneHeader')}
          icon={'phone'}
        />
        <Card
          title={t('MainHomeScreen.sidebar.details.email')}
          heading={t('MainHomeScreen.sidebar.details.emailHeader')}
          icon={'email'}
        />
        <Card
          title={t('MainHomeScreen.sidebar.details.location')}
          heading={t('MainHomeScreen.sidebar.details.locationHeader')}
          icon={'location'}
        />
        <Card
          title={t('MainHomeScreen.sidebar.details.birthday')}
          heading={t('MainHomeScreen.sidebar.details.birthdayHeader')}
          icon={'birthday'}
        />
      </Box>
      <Box
        display={'flex'}
        borderRadius={'lg'}
        boxShadow={'lg'}
        marginX={'4'}
        padding={'2'}
        justifyContent={'center'}
        bg={bg.container}
      >
        <Text fontSize={'lg'} fontWeight={'bold'}>
          {t('MainHomeScreen.sidebar.post')}
        </Text>
      </Box>
      <Box position="relative" padding="10" top={-5}>
        <Divider />
        <AbsoluteCenter bg={bg.container} borderRadius={'lg'} px={4}>
          Social links
        </AbsoluteCenter>
      </Box>
      <SocialButtons />
      <Text display={'flex'} justifyContent={'center'}>
        {t('MainHomeScreen.sidebar.version') + packageJson.version}
      </Text>
    </Box>
  );
};

export default Content;
