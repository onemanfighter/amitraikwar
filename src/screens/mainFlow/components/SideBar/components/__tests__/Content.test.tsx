import { render } from '@testing-library/react';
import Content from '../Content';
import { ThemeProvider } from '@providers';

jest.mock('react-social-icons', () => ({
  SocialIcon: ({ url }: { url: string }) => <div>{url}</div>,
}));

describe('Content', () => {
  it('renders correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <Content />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
