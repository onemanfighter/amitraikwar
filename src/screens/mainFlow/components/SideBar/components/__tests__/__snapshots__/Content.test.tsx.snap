// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="light",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <div
    class="css-1yspfdw"
  >
    <div
      class="css-12f1a2y"
    >
      <img
        class="chakra-image css-10ttzh9"
        loading="lazy"
        src=""
      />
    </div>
    <div
      class="css-16mslk0"
    >
      <span
        class="index-module_type__E-SaG"
        style="font-size: 1.3em; font-weight: bold; font-family: monospace; text-align: center;"
      />
    </div>
    <div
      class="css-1n7u47g"
    >
      <div
        class="css-1t5din3"
      >
        <button
          aria-label="raikwar.amit.1603@gmail.com"
          class="chakra-button css-1u36519"
          type="button"
        >
          <svg
            class="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 8.67v8.58a3 3 0 0 0 3 3h15a3 3 0 0 0 3-3V8.67l-8.928 5.493a3 3 0 0 1-3.144 0L1.5 8.67Z"
            />
            <path
              d="M22.5 6.908V6.75a3 3 0 0 0-3-3h-15a3 3 0 0 0-3 3v.158l9.714 5.978a1.5 1.5 0 0 0 1.572 0L22.5 6.908Z"
            />
          </svg>
        </button>
        <div
          class="css-o99mrc"
        >
          <p
            class="chakra-text css-itvw0n"
          >
            Email
          </p>
          <p
            class="chakra-text css-avjjns"
          >
            raikwar.amit.1603@gmail.com
          </p>
        </div>
      </div>
      <div
        class="css-1t5din3"
      >
        <button
          aria-label="India"
          class="chakra-button css-1u36519"
          type="button"
        >
          <svg
            class="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="m11.54 22.351.07.04.028.016a.76.76 0 0 0 .723 0l.028-.015.071-.041a16.975 16.975 0 0 0 1.144-.742 19.58 19.58 0 0 0 2.683-2.282c1.944-1.99 3.963-4.98 3.963-8.827a8.25 8.25 0 0 0-16.5 0c0 3.846 2.02 6.837 3.963 8.827a19.58 19.58 0 0 0 2.682 2.282 16.975 16.975 0 0 0 1.145.742ZM12 13.5a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          class="css-o99mrc"
        >
          <p
            class="chakra-text css-itvw0n"
          >
            Location
          </p>
          <p
            class="chakra-text css-avjjns"
          >
            India
          </p>
        </div>
      </div>
    </div>
    <div
      class="css-n0xb99"
    >
      <p
        class="chakra-text css-1xs2h3i"
      >
        Software Engineer 3 @ RSLabs
      </p>
    </div>
    <div
      class="css-11avr8y"
    >
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-14ww7qh"
      />
      <div
        class="css-1roiji1"
      >
        Social links
      </div>
    </div>
    <div
      class="css-17lwykd"
    >
      <div
        class="css-pocmpu"
      >
        <div>
          https://in.linkedin.com/in/amitrai1603
        </div>
      </div>
      <div
        class="css-pocmpu"
      >
        <div>
          https://github.com/onemanfighter
        </div>
      </div>
      <div
        class="css-pocmpu"
      >
        <div>
          https://github.com/raikwaramit
        </div>
      </div>
      <div
        class="css-pocmpu"
      >
        <div>
          https://amitraikwar.medium.com/
        </div>
      </div>
    </div>
    <p
      class="chakra-text css-1l4w6pd"
    >
      Version: 0.2.0
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
